config/secret.go

```diff
@@ -81,6 +88,11 @@ func (s *Secret) Destroy() {
 	s.enclave = nil
 }
 
+// Empty return if the secret is completely empty
+func (s *Secret) Empty() bool {
+	return !s.notempty
+}
+
```

Comment:
If we are going to have a boolean variable + boolean function, shouldn't they be of the same value (i.e. `empty` + `Empty()`)?

Reply:
Well I tend to agree, and that's what I had first, but there is a subtle case making this necessary. Think about an config option that is not specified in TOML. In this case, the secret will be created (in the sense of the instance exists) in the plugin, but it will not be initialized _at all_. This leads to the default value for its members rendering an `empty` option to `false`. To mark uninitialized secret as empty (which is equivalent to how strings are seen) I simply inverted the option. Of course you can turn around the whole function (e.g. name it `Filled()`), but the usual use-case in the plugins is to check for empty (e.g. string) and then set some sort of default, which means you would have a `if !myplugin.SecretStuff.Filled()`...